modApiServer/test/org/aion/api/server/ApiAionTests.java

```diff
@@ -0,0 +1,398 @@
+            evBlock.setFuncArgs(Collections.singletonList(abs));
+
+            ees.add(evBlock);
+
+            //provokes exception in EpApi.run()
+            ees.add(new EventBlock(EventBlock.CALLBACK.ONBLOCK0));
+            ees.add(new EventDummy());
+        }
+
+    }
+
+    @Test
+    public void TestCreate() {
+        System.out.println("run TestCreate.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        System.out.println("API Version = " + api.getApiVersion());
+        assertNotNull(api.getInstalledFltrs());
+    }
+
+    @Test
+    public void TestInitNrgOracle() {
+        System.out.println("run TestInitNrgOracle.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+        // Initing a second time should not create a new NrgOracle
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+    }
+
+    @Test
+    public void TestStartES() {
+        System.out.println("run TestStartES.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.startES("thName");
+        api.addEvents();
+        try
+        {
+            Thread.sleep(5000);
+        }
+        catch(InterruptedException ex) {}
+        api.shutDownES();
+        assertTrue(api.allFlagsSet());
+    }
+
+    @Test
+    public void TestGetBlock() {
+        System.out.println("run TestGetBlock.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        assertNotNull(api.getBlockTemplate());
+
+        AionBlock blk = impl.getBlockchain().getBestBlock();
+
+        assertEquals(blk, api.getBestBlock());
+
+        assertEquals(blk, api.getBlock(-1));
+
+        assertEquals(blk.toString(), api.getBlockByHash(blk.getHash()).toString());
+        assertEquals(blk.toString(), api.getBlock(blk.getNumber()).toString());
+
+        Map.Entry rslt = api.getBlockWithTotalDifficulty(blk.getNumber());
+        assertEquals(rslt.getKey().toString(), blk.toString());
+        if (!blk.isGenesis())
+            assertEquals(rslt.getValue(),
+                    ((AionBlockStore)impl.getBlockchain().getBlockStore()).getTotalDifficultyForHash(blk.getHash()));
+        assertEquals(api.getBlockWithTotalDifficulty(0).getValue(), CfgAion.inst().getGenesis().getDifficultyBI());
+    }
+
+
+    @Test
+    public void TestGetSync() {
+        System.out.println("run TestGetSync.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        SyncInfo sync = api.getSync();
+        assertNotNull(sync);
+        assertEquals(sync.done, impl.isSyncComplete());
+        if (impl.getInitialStartingBlockNumber().isPresent())
```

Comment:
Same question about the if statements

Reply:
they're optionals, so it just felt like good practice to check that there's a value present before poking it